### DISCUSSION TITLE
Installation problem On HPC

### DISCUSSION THREAD

**Discussion by zxj19910214** [idaholab/moose#16609] [Q&A Getting Started] [answered]
Dear Everyone,

Happy new year!

My name is Xiaojing, I am trying to install moose on our HPC, but I got 14 tests failed, I can still generate my executable(opt) files. I am writing to ask whether the failing tests will influence my simulation performance.
you can see the failed process in the following picture:
![image](https://user-images.githubusercontent.com/38807201/103521294-cdc1bc00-4e78-11eb-89c6-2d12ea1a0031.png)

Warm regards,
Xiaojing

**Comment by aeslaughter** [comment]
If you scroll up in your window you will have the complete error messages for each test. Take a look at those error messages and see if there is something consistent between them and share that information.

**Comment by zxj19910214** [comment]
Thank you so much for your reply, I added my test-log file below.

[log-tests.txt](https://github.com/idaholab/moose/files/5766315/log-tests.txt)

**Reply by GiudGiud** [reply]
There's a variety of errors. Looks like you don't have matplotlib installed. 
There's some MPI and Petsc errors too. How did you install petsc ?

**Reply by zxj19910214** [reply]
Thank you for your answer, I noticed there is a lot of errors after I install Moose on HPC.  I am using a system-wide installation provided by the HPC team.  They download petsc(3.11.1) from: https://ftp.mcs.anl.gov/pub/petsc/release-snapshots/
and install it.  Will these errors influence the performance of the simulation? 

Because I tried to use partitioner in my code: https://mooseframework.inl.gov/syntax/Mesh/Partitioner/index.html#!

It seems the performance is not improving when I add a partitioner. I feel curious about whether I did something wrong when I install moose. Hope you can give me some suggestions on how to figure this out. 

Thank you for your help

**Comment by moravveji** [comment, marked as answer]
Hi.
I have installed MOOSE on our machine so that our users can compile their own work in their local directories. To build MOOSE, I use the pre-installed PETSc module; everything is compiled (consistently?) using FOSS 2018a toolchain (i.e. with OpenMPI 2.1.2). However, I have the flexibility to recompile MOOSE (and its dependencies) using more recent toolchains, if that might make a performance difference. 
I have attached the build script used for the current version, so, I appreciated if you give a look, and let me know if anything can be improved.
[moose.zip](https://github.com/idaholab/moose/files/5780077/moose.zip)

**Reply by moravveji** [reply]
Additions: 

- For your missing matplotlib dependency (On Tier-1 skylake partition), you can use `matplotlib/3.1.1-intel-2018a-Python-3.7.2` module, which is compatible with the Python dependency version
-  In the error log you sent, I see the error: `PETSC ERROR: Could not locate solver package superlu_dist. Perhaps you must ./configure with --download-superlu_dist`. I suspected that SUPERLU was missing from PETSc package
- I inspected the build logs of the used module `PETSc/3.11.1-foss/2018a`, and searched for results of SUPERLU test; Indeed, SUPERLU is installed as part of PETSc, and is tested during the build time; see e.g. `/apps/leuven/skylake/2018a/software/PETSc/3.11.1-foss-2018a/share/petsc/examples/src/snes/examples/tutorials/output/ex19_superlu.out`
- The same is true for the Cholesky solver
- I checked the group permissions on the PETSc folder, and I realized that it needed some fix.

Would you please try again with your tests?

**Reply by zxj19910214** [reply]
Thank you very much for your help, i will test it and see whether it helps
